src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSpanClearInsteadOfFill.cs

```diff
@@ -0,0 +1,150 @@
+            DiagnosticId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            description: s_localizableDescription,
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(s_Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var compilation = context.Compilation;
+            if (!compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemSpan1, out var spanType))
+            {
+                return;
+            }
+
+            var spanFillMethod = spanType.GetMembers(FillMethod)
+                .OfType<IMethodSymbol>()
+                .FirstOrDefault();
+
+            if (spanFillMethod?.Parameters.Length != 1)
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(
+                context =>
+                {
+                    var invocation = (IInvocationOperation)context.Operation;
+
+                    if (SymbolEqualityComparer.Default.Equals(invocation.TargetMethod.OriginalDefinition, spanFillMethod)
+                        && IsDefaultValue(invocation.Arguments[0]))
+                    {
+                        context.ReportDiagnostic(invocation.CreateDiagnostic(s_Rule));
+                    }
+                },
+                OperationKind.Invocation);
+        }
+
+        private static bool IsDefaultValue(IArgumentOperation argumentOperation)
+        {
+            var value = argumentOperation.Value;
+            var type = argumentOperation.Parameter.Type;
+            var constantOpt = value.ConstantValue;
+
+            if (constantOpt.HasValue)
+            {
+                if (constantOpt.Value == null)
+                {
+                    // null must be default value for any valid type
+                    return true;
+                }
+
+                if (argumentOperation.Type.IsNullableValueType())
+                {
+                    // 0 isn't default value for T?
+                    return false;
+                }
+
+                // enum/nint are treated as integers
+                // This is missing default value of DateTime literal for VB,
+                // but since VB doesn't properly support Span, just don't consider it
+                switch (constantOpt.Value)
+                {
+#pragma warning disable IDE0004 // Remove Unnecessary Cast - false positive
```

Comment:
Can we file an issue on Roslyn repo for this false positive? I wonder if the false positive is now resolved as @CyrusNajmabadi has completely rewritten remove unnecessary cast analyzer.

Reply:
It still repros at VS 17.4P1. I remember the rewritten was delayed?

Reply:
@huoyaoyuan https://github.com/dotnet/roslyn/pull/56932

Reply:
The rewrite was shipped in 17.1. But the code style packages used in this repo are 4.0

https://github.com/dotnet/roslyn-analyzers/blob/793113a41d7f21b03470521bf48438f2abd9b12f/eng/Versions.props#L42-L44

So we are not yet consuming the fix from https://github.com/dotnet/roslyn/pull/56932

Reply:
@mavasani do we want to update this version with this PR? Seems it is only open question in this PR

Reply:
@buyaa-n Updating requires more changes. I'm handling that in another PR.

Reply:
OK, just making sure all comments resolved